api/src/services/fields.ts

```diff
@@ -764,7 +764,17 @@ export class FieldsService {
 						.where({ group: metaRow.field, collection: metaRow.collection });
 				}
 
-				await trx('directus_fields').delete().where({ collection, field });
+				const fieldIds = (await trx('directus_fields').select('id').where({ collection, field })).map(
+					(field) => field.id,
+				);
+
+				const itemsService = new ItemsService('directus_fields', {
+					knex: trx,
+					accountability: this.accountability,
+					schema: this.schema,
+				});
+
+				await itemsService.deleteMany(fieldIds, { emitEvents: false });
```

Comment:
Could we simplify this with a 

```js
itemsService.deleteByQuery({
  filter: {
    collection: { _eq: collection },
    field: { _eq: field }
  }
})
```

?

Reply:
Absolutely~ 💯 Thank you!
Patched in https://github.com/directus/directus/pull/24188/commits/145aa46e4a973689f9db2b5a24beb55020c9933c